src/main/java/models/Storage.java

```diff
@@ -0,0 +1,132 @@
+package models;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+public class Storage {
+    private final String filePath;
+    private final String directoryPath;
+
+    public Storage(String filePath, String directoryPath) {
+        this.filePath = filePath;
+        this.directoryPath = directoryPath;
+    }
+
+    /**
+     * Attempt to retrieve a local save of the user's tasks on their pc as a list, if not found,
+     * return the an empty list
+     * 
+     * @return List containing either existing Todos based on data file or an empty list
+     */
+    public List<Optional<? extends Todo>> retrieveLocalDatabase() {
+        try {
+            // get local file
+            File localFile = new File(this.filePath);
+            Scanner sc = new Scanner(localFile);
+            List<Optional<? extends Todo>> existingTodosList = new ArrayList<>();
+            while (sc.hasNextLine()) {
+                // get line, splitting by special character delimiter |
+                List<String> line = Arrays.asList(sc.nextLine().split("\\|"));
+
+                // line = [type, isDone, message, extraMessage (event / deadline)]
+                String type = line.get(0);
+                boolean isDone = line.get(1) == "1";
```

Comment:
Maybe put a bracket around `line.get(1) == "1"` to make it easier to be understood